# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Cell.h
# Commit: 43978ebfc7ac
# Full Hash: 43978ebfc7ace2c37aba0b7644ac4e866e0730bd
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2021-01-20 16:13:57
# Regressor Bug: 1686219
# File Overlap Count: 1
# Description:
#   Bug 1686219 - Store chunk information at the start rather than the end of the chunk r=sfink
#   
#   This moves the chunk metadata to the start of the chunk and defines the data structures in the public header.  This simplifies accessing this data and removes the need for hardcoded offsets.
#   
#   Requesting review from jandem for JIT updates.
# ==============================================================================

diff -r 496d47ff3fa1 -r 43978ebfc7ac js/src/gc/Cell.h
--- a/js/src/gc/Cell.h	Wed Jan 20 10:04:11 2021 +0000
+++ b/js/src/gc/Cell.h	Wed Jan 20 10:04:29 2021 +0000
@@ -330,13 +330,13 @@
 }
 
 inline JSRuntime* Cell::runtimeFromMainThread() const {
-  JSRuntime* rt = chunk()->trailer.runtime;
+  JSRuntime* rt = chunk()->header.runtime;
   MOZ_ASSERT(CurrentThreadCanAccessRuntime(rt));
   return rt;
 }
 
 inline JSRuntime* Cell::runtimeFromAnyThread() const {
-  return chunk()->trailer.runtime;
+  return chunk()->header.runtime;
 }
 
 inline uintptr_t Cell::address() const {
@@ -354,7 +354,7 @@
 }
 
 inline StoreBuffer* Cell::storeBuffer() const {
-  return chunk()->trailer.storeBuffer;
+  return chunk()->header.storeBuffer;
 }
 
 JS::Zone* Cell::zone() const {